wrappers/c/zxing-c.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+* Copyright 2023 siiky
+* Copyright 2023 Axel Waggershauser
+*/
+// SPDX-License-Identifier: Apache-2.0
+
+#include "zxing-c.h"
+
+#include "ReadBarcode.h"
+
+using namespace ZXing;
+
+char* copy(std::string_view sv)
+{
+	auto ret = (char*)malloc(sv.size() + 1);
+	if (ret) {
+		strncpy(ret, sv.data(), sv.size());
+		ret[sv.size()] = '\0';
+	}
+	return ret;
+}
+
+extern "C"
+{
+	/*
+	 * ZXing/ImageView.h
+	 */
+
+	zxing_ImageView* zxing_ImageView_new(const uint8_t* data, int width, int height, zxing_ImageFormat format, int rowStride,
+										 int pixStride)
+	{
+		ImageFormat cppformat = static_cast<ImageFormat>(format);
+		return new ImageView(data, width, height, cppformat, rowStride, pixStride);
+	}
+
+	void zxing_ImageView_delete(zxing_ImageView* iv)
+	{
+		delete iv;
+	}
+
+	int zxing_ImageView_width(const zxing_ImageView* iv)
+	{
+		return iv->width();
+	}
+
+	int zxing_ImageView_height(const zxing_ImageView* iv)
+	{
+		return iv->height();
+	}
+
+	int zxing_ImageView_pixStride(const zxing_ImageView* iv)
+	{
+		return iv->pixStride();
+	}
+
+	int zxing_ImageView_rowStride(const zxing_ImageView* iv)
+	{
+		return iv->rowStride();
+	}
+
+	zxing_ImageFormat zxing_ImageView_format(const zxing_ImageView* iv)
+	{
+		return static_cast<zxing_ImageFormat>(iv->format());
+	}
+
+	/*
+	 * ...
+	 */
+
+	/*
+	 * ZXing/BarcodeFormat.h
+	 */
+
+	zxing_BarcodeFormats zxing_BarcodeFormatsFromString(const char* str)
+	{
+		if (!str)
+			return {};
+		try {
+			auto format = BarcodeFormatsFromString(str);
+			return static_cast<zxing_BarcodeFormats>(*reinterpret_cast<BarcodeFormat*>(&format));
+		} catch (...) {
+			return zxing_BarcodeFormat_Invalid;
+		}
+	}
```

Comment:
I saw you replaced `zxing_BarcodeFormatFromString` with `zxing_BarcodeFormatsFromString`. Initially I implemented the singular because I didn't know how to implement the plural (it looked like `zxing_BarcodeFormats` was slightly more complicated than just a `zxing_BarcodeFormat` typedef), but also because `zxing_BarcodeFormatFromString` is more useful to us to prevent (programmer) mistakes -- we only need to scan one format at a time.

Is it ok to have both, or are you against it?

Reply:
I'm fine with it and just added it...